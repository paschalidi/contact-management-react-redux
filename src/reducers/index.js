import { combineReducers } from 'redux';
import { reducer as formReducer } from 'redux-form';
import reducerAddDeleteContact from './reducer-add-delete-contact'

const rootReducer = combineReducers({
  form: formReducer,
  reducerAddDeleteContact
});

export default rootReducer;
