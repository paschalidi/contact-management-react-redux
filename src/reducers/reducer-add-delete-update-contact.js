import {
  ADD_UPDATE_CONTACT,
  DELETE_CONTACT,
} from '../actions/types'

const INITIAL_STATE = {
  0: {
    id: null,
    firstName: null,
    lastName: null,
    email: null
  }
};

export default function (state = INITIAL_STATE, action) {
  switch (action.type) {
    case ADD_UPDATE_CONTACT:
      const { id, firstName, lastName, email } = action.payload;

      return Object.assign(
        {},
        state,
        { [id]: { id, firstName, lastName, email } }
      );
    case DELETE_CONTACT:
      const { identifier } = action.payload;
      let newState = Object.assign({}, state);

      delete newState[identifier];
      return newState;

    default:
      return state;
  }
}