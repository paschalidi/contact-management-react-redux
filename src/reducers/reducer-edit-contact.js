import {
  EDIT_CONTACT
} from '../actions/types'

const INITIAL_STATE = {
  id: null,
  firstName: null,
  lastName: null,
  email: null
};

export default function (state = INITIAL_STATE, action) {
  switch (action.type) {
    case EDIT_CONTACT:
      return {
        data: action.payload
      };
    default:
      return state;
  }
}