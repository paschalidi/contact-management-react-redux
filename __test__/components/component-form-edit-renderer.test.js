import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';

import reducers from '../../src/reducers';
import EditContact from '../../src/components/component-form-edit-renderer'

import renderer from 'react-test-renderer';

describe('CardComponent', () => {
  const createStoreWithMiddleware = applyMiddleware()(createStore);
  it('renders correctly', () => {
    const component = renderer.create(
      <Provider store={createStoreWithMiddleware(reducers)}>
        <EditContact />
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});