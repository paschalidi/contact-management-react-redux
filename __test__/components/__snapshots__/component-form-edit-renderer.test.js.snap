// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent renders correctly 1`] = `
<div
  className="container-edit-contact"
>
  <div
    className="col-md-6"
  >
    <div
      className="panel panel-default"
    >
      <div
        className="panel-heading"
      >
        <h3
          className="panel-title"
        >
          Edit Contact
        </h3>
      </div>
      <div
        className="panel-body"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label>
              First Name
            </label>
            <div>
              <input
                className="form-control"
                name="firstName"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="First Name"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="form-group"
          >
            <label>
              Last Name
            </label>
            <div>
              <input
                className="form-control"
                name="lastName"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="Last Name"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="form-group"
          >
            <label>
              mail
            </label>
            <div>
              <input
                className="form-control"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="mail"
                type="text"
                value=""
              />
            </div>
          </div>
          <div>
            <button
              className="btn btn-success"
              disabled={false}
              type="submit"
            >
              Update
            </button>
            <button
              className="btn btn-default"
              disabled={true}
              onClick={[Function]}
              type="button"
            >
              Undo Changes
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
